presto-main/src/main/java/com/facebook/presto/operator/DistinctLimitOperator.java

```diff
@@ -186,16 +203,50 @@ public void addInput(Page page)
                 }
             }
         }
+        groupByIds = null;
+        inputPage = null;
+
+        Page result = null;
         if (!pageBuilder.isEmpty()) {
-            outputPage = pageBuilder.build();
+            result = pageBuilder.build();
+            pageBuilder.reset();
```

Comment:
previously we were calling `pageBuilder.reset();` unconditionally in `addInput` after setting the memory reservation, is it correct to move that call to here and do it in a conditional way?

Reply:
I checked the logic of the code and made the change. Seems `pageBuilder` only serves for building an output page for every `getOutput` call. So I added `verify(pageBuilder.getPositionCount() == 0);` at the beginning of this function. It would be good if multiple people can verify the logic.